e2e/test/scenarios/admin/datamodel/reproductions/36422-show-create-metric-error.cy.spec.js

```diff
@@ -0,0 +1,28 @@
+import { popover, restore } from "e2e/support/helpers";
+
+describe("issue 36422", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should show error and disable save button when create metric fails", () => {
+    cy.visit("/admin/datamodel/metrics");
+    cy.button("New metric").click();
+    selectTable("Orders");
+    cy.get('[name="name"]').type("x");
+    cy.get('[name="description"]').type("x");
+
+    cy.intercept("POST", "/api/metric", req => req.reply(400));
+    cy.button("Save changes").click();
+    cy.button("Failed")
+      .parent()
+      .parent()
+      .should("contain", "An error occurred");
```

Comment:
Same here. Ideally there is a role we can use to grab the status

Reply:
Ran the test locally. I'd consider adding a role to the status text, or just adding a test id to the form footer on the page